### ISSUE TITLE
508-defect-3 [FOCUS MANAGEMENT]: Focus order SHOULD flow in the logical reading order

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9374]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On a **Yellow Ribbon** page of search results, the focus order currently flows from the Search button to the links below. Content order **should** flow from the Search button to the title above the search results, then the links below the search button, and then the footer links.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate the page by tabbing and experience a meaningful content flow.

## Environment

* Operating System: all
* Browser: all
* Server destination: production

## Steps to Recreate

1. Enter https://www.va.gov/education/yellow-ribbon-participating-schools/?state=AL
 in browser
2. Tab through the page
3. Verify that the keyboard focus moves from the Search button to the links beneath it

### Content order
I checked with @andaleliz before documenting this bug, and validated this solution.

The content order should flow from the Search button and then to:
1. heading introducing the search results
1. through the individual search results
1. search results pagination
1. Helpful Links
1. footer

Edit: in order for this to be structured appropriately by headings, helpful links should be placed below the search results pagination 

### Recommendation (Josh's edit)
Ideally, the best fix would be to move the visual AND DOM location of "helpful links" to where it is placed in the content order above. In other words, It should be right below the search results pagination. (see screenshot below)

<img width="1338" alt="Screen Shot 2021-08-17 at 9 59 45 AM" src="https://user-images.githubusercontent.com/14154792/129739699-a0b6e848-657a-421f-a2e2-c8858f83d32f.png">

If moving the visual location of "helpful links" is not possible, the next alternative would be to use CSS to visually reposition it while keeping its location in the DOM right below the search results pagination.

If moving the DOM location of "helpful links" is not possible, I would recommend keeping this ticket open until a better solution can be found. I strongly recommend against using a positive `tabindex` to fix this.

#### Heading adjustments
<img width="1179" alt="Screen Shot 2021-08-17 at 10 38 00 AM" src="https://user-images.githubusercontent.com/14154792/129746580-29c495f9-c5d0-4509-a204-82668a2db6bf.png">

1. All search result titles (university names) should be `h3`s
2. Helpful links should be a `h2`

## WCAG or Vendor Guidance (optional)

* [W3C WCAG S.C. 1.3.2: Meaningful Sequence](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html)

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO This could be a potential worm hole. Going to give it at least a 5 est.

**Comment by MarciMcGuireGCIO** [comment]
Thanks, @zacharymorel !

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO When you get back, Josh and I have done some in depth conversation on this ticket. 
We'd like to rope in @callen2563 for a new design and placement of the "helpful links" section! Thanks

**Comment by MarciMcGuireGCIO** [comment]
Looping in @callen2563 and @rtwell - can either of you think of a reason why we could not make the change that Josh describes above (moving Helpful links below pagination)?  Thanks!

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel, @callen2563 has now done a mockup for this - please see https://github.com/department-of-veterans-affairs/va.gov-team/issues/29244.

**Comment by zacharymorel** [comment]
Perfect @MarciMcGuireGCIO !

**Comment by zacharymorel** [comment]
@joshkimux This is ready for review!

**Comment by joshkimux** [comment]
@zacharymorel Brilliant! Positioning in the DOM is much better for accessibility 🥳 
One last little detail-- could we update the heading level for "Helpful links" to a `h2` instead of a `h3`?

**Comment by zacharymorel** [comment]
> @zacharymorel Brilliant! Positioning in the DOM is much better for accessibility 🥳
> One last little detail-- could we update the heading level for "Helpful links" to a `h2` instead of a `h3`?

Yes! That makes sense for flow of page.

**Comment by zacharymorel** [comment]
@joshkimux Give it a whirl on staging

**Comment by joshkimux** [comment]
<img width="1621" alt="Screen Shot 2021-09-21 at 2 17 37 PM" src="https://user-images.githubusercontent.com/14154792/134225665-9745e6b7-9ca1-40c1-a442-cd622187d8c1.png">
Whirled it and it looks great! Thanks @zacharymorel for the quick work 🥳
